src/components/Search/Search.tsx

```diff
@@ -83,26 +81,27 @@ const Search: VibeComponent<SearchProps, unknown> & {
       searchResultsContainerId = "",
       activeDescendant = "",
       iconNames = SearchDefaultIconNames,
-      loading = false
+      loading = false,
+      "data-testid": dataTestId
     },
     ref
   ) => {
     const mergedRef = useMergeRefs({ refs: [ref, setRef] });
     return (
       <TextField
         id={id}
-        dataTestId={getTestId(ComponentDefaultTestId.SEARCH, id)}
+        dataTestId={dataTestId || getTestId(ComponentDefaultTestId.SEARCH, id)}
         iconName={iconName}
         value={value}
         onChange={onChange}
         autoFocus={autoFocus}
         placeholder={placeholder}
         disabled={disabled}
         debounceRate={debounceRate}
-        className={classNames(className, "search_component", getType(type))}
+        className={cx(className, styles.search, "search_component", getType(type))}
         secondaryIconName={secondaryIconName}
         secondaryDataTestId={getTestId(ComponentDefaultTestId.CLEAN_SEARCH_BUTTON, id)}
```

Comment:
maybe `secondaryDataTestId` prop is needed here?

Reply:
didn't add because was not sure about the syntax of the prop, this pr is already huge let's decide it separately